hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java

```diff
@@ -125,24 +127,38 @@ public StateView(
 	}
 
 	public Optional<FileGetInfoResponse.FileInfo> infoForFile(FileID id) {
-		try {
-			var attr = fileAttrs.get(id);
-			if (attr == null) {
+		int retries = RETRY_TIMES;
+		while (true) {
```

Comment:
I would recommend not using while true loops. I suggest to move this try block to a method that throws `BinaryObjectNotFoundException` and we catch it here and retry.

Reply:
Done with that. Still need a loop to retry, though not while(true) loop.